crates/rmf_site_editor/src/widgets/tasks/mod.rs

```diff
@@ -337,481 +289,793 @@ fn show_task(
         .default_open(false)
         .show(ui, |ui| {
-            Grid::new("edit_task_details")
-                .num_columns(2)
-                .show(ui, |ui| {
-                    let mut new_task_params = task_params.clone();
-
-                    // Start time
-                    ui.label("Start Time:")
-                        .on_hover_text("(Optional) The earliest time that this task may start");
-                    let start_time = new_task_params.start_time();
-                    let mut has_start_time = start_time.is_some();
-                    ui.horizontal(|ui| {
-                        ui.checkbox(&mut has_start_time, "");
-                        if has_start_time {
-                            let new_start_time = new_task_params.start_time_mut().get_or_insert(0);
-                            ui.add(
-                                DragValue::new(new_start_time)
-                                    .range(0_i32..=std::i32::MAX)
-                                    .speed(1),
-                            );
-                        } else if start_time.is_some() {
-                            *new_task_params.start_time_mut() = None;
-                        }
-                    });
-                    ui.end_row();
-
-                    // Request time
-                    ui.label("Request Time:")
-                        .on_hover_text("(Optional) The time that this request was initiated");
-                    let request_time = new_task_params.request_time();
-                    let mut has_request_time = request_time.is_some();
-                    ui.horizontal(|ui| {
-                        ui.checkbox(&mut has_request_time, "");
-                        if has_request_time {
-                            let new_request_time =
-                                new_task_params.request_time_mut().get_or_insert(0);
-                            ui.add(
-                                DragValue::new(new_request_time)
-                                    .range(0_i32..=std::i32::MAX)
-                                    .speed(1),
-                            );
-                        } else if request_time.is_some() {
-                            *new_task_params.request_time_mut() = None;
-                        }
-                    });
-                    ui.end_row();
+            // Fleet name
+            if task.is_dispatch() {
+                ui.horizontal(|ui| {
+                    ui.label("Fleet:").on_hover_text(
+                        "(Optional) The name of the fleet for this robot. \
+                            If specified, other fleets will not bid for this task.",
+                    );
+                    if !in_edit_mode {
+                        ui.label(task_request.fleet_name().unwrap_or("None".to_string()));
+                    } else {
+                        edit_fleet_widget(ui, &mut new_task);
+                    }
```

Comment:
I noticed that on my machine "some" tooltips don't work, specifically `Fleet` doesn't but `Requester` does, can you reproduce?
I also wonder since we are pre-populating the fleet, do we actually want the fleet parameter to be editable or should it just be fixed?
Making it editable creates a risk of disconnect, for example `TinyRobot1` is spawned in the world as part of `Fleet1`, users decide to assign a direct task to it and choose it from the dropdown, the fleet is automatically initialized to `Fleet1`, which is great.
However, the field is editable so users can change the fleet name to `Fleet2` and still create a valid task but it seems to me that is an invalid task that should not be allowed.
What do you think?

Reply:
Yeah I think you're right, it was quite an inconsistent design. Though now we do not have any other way to modify a robot's fleet, which was the intention of the text edit UI. I'll add one to the robot properties widget and make this non-editable.